### ISSUE TITLE
ZWave device not properly recognised in OH2 (but recognised OK in OH1)

### ISSUE THREAD

**Issue by roydonm** [openhab/openhab2-addons#1397]
I was running Openhab 1 using ZWave binding snapshot 1.9.0 and it was properly recognising my ZWave Fibaro light sensor (FGS212) without any issues.  I am running openhab 2 on a seperate machine and when all the 'things' are identified, that node is not being properly recognised.  The details that Openhab 2 is showing for that node re very different from Openhab 1 (different manufacturer code, different device ID, etc).

I installed openhab 2 from the respository (apt-get) and I have no idea how to update the ZWave binding (as maybe the binding that came with the repository is a bit outdated).  I will be happy to try that is someone can tell me where to find the latest snapshot of teh ZWave binding for openhab 2, and also where I must put that file on my Pi (I am used to openhab 1 on windows, not openhab 2 on linux...).

**Comment by roydonm** [comment]
Sorry, meant to say my device is FGS212.  This is what is showing in Habmin (which is not the correct manufacturer ID, etc):

![image](https://cloud.githubusercontent.com/assets/10046961/20100801/c355ed58-a5c7-11e6-9cb7-99f8b50baca4.png)


I am not sure where these details came from, as OH1 displays the correct details and works perfectly with the device, it is only doing this in OH2 (with the same Zwave controller).

**Comment by roydonm** [comment]
I also found that the binding I have installed is 2.0.0.b4:

/usr/share/openhab2/runtime/karaf/system/org/openhab/binding/org.openhab.binding.zwave/2.0.0.b4/org.openhab.binding.zwave-2.0.0.b4.jar

**Comment by cdjackson** [comment]
I've added the type/Id to the database. You will need to update to a new version of the binding.

Closing this as its not the zwave issue tracker (there's a separate repo for zwave issues).

**Comment by roydonm** [comment]
Chris,  FIrstly, thanks !!

Secondly, is there a link that describes:
1 - Where to get the most recent version of the zwave binding, and
2 - How to install it in OH2

I've been searching for hours independently on how to find this for OH2 and I can't find anything.  OH1 on Windows was very simple for me, but with OH2, I have no idea where to find the latest binding builds and how to install them (i.e. which folder do I drop it in) ?

**Comment by cdjackson** [comment]
I'm not sure about updated through apt, but you can get the latest snapshot from cloudbees -:

https://openhab.ci.cloudbees.com/job/openHAB-Distribution/

Once you are using the snapshot version, then you can update a binding simply by uninstalling it, and reinstalling it again (ie through PaperUI or HABmin - when it installs a snapshot, it will grab the latest version).

If you use an offline version, or a specific version (eg the beta) then it will always install that specific version.

**Comment by ThomDietrich** [comment]
I'm not sure if this will work as you said you are on beta 4:
Connect to the karaf console, type `bundle:list` and find the id of the zwave binding, execute `bundle:update ID`. `bundle:list` should show the new version time stamp.

**Comment by cdjackson** [comment]
This won't work - if you use an offline version, or a specific version (eg beta4) then it will always install that specific version.

**Comment by BClark09** [comment]
You can pull the compiled jar from the build and update with bundle:update [ID] file://[LOCATION] though, it's been a while since I moved to the online snapshot, but I recall this working for me.

**Comment by roydonm** [comment]
OK, I manage to get the latest snapshot installed and it's working like a peach :)  Thanks to Chris for the awesome work he's doing on Wave with Openhab, my house would be a different (and worse) palce if it wasn't for his efforts !

Just in case anyone struggles the way I have with finding and installing a latest version of a binding, I've captured the steps I went through here:
1 - I went to https://openhab.ci.cloudbees.com/job/openHAB2-Bundles/ and then clicked on 'Workspace' and then followed the links 'bindings/org.openhab.binding.zwave' -> 'target'.  There I found the JAR file called 'org.openhab.binding.zwave-2.0.0-SNAPSHOT.jar' which I downloaded.
2 - On my RPi I went to the folder '/usr/share/openhab2/runtime/karaf/system/org/openhab/binding/org.openhab.binding.zwave/2.0.0.b4' and in there I found the ZWave JAR file that was currently being used by my version of OH2.  I removed that file and copied into that folder the file that I downloaded in step 1 above.  Note that I did have permission issues so I did have to 'chmod 777' the folder '2.0.0.b4' as I was working as the user 'pi'.
3 - I stopped and started OH2

My FGS212 was now recognised (again, thanks to the super-fast update from Chris) and all is well again.  My RPi is running OH2 without issue, although I will say that the reaction time of the ZWave network is not as fast as it was on my Windows server running OH1.  I'm hoping the upside of running it on a tiny RPi makes up for that 5 second extra wait time each time I turn a light on/off.
